src/index.ts

```diff
@@ -1 +1,525 @@
+    /**
+     * Start the app.
+     *
+     * @see {@link PlatformApi}
+     */
+    startApp: () => Promise<void>;
+
+    /**
+     * Start collecting the device's traffic. This will start a proxy on the host computer on port `8080`. You need to
+     * configure your device to use this proxy (unless you are using an Android emulator).
+     *
+     * Only one traffic collection can be active at a time.
+     *
+     * @param name An optional name to later identify this traffic collection, defaults to the current date otherwise.
+     *
+     * @todo Automatically configure the device to use the proxy (https://github.com/tweaselORG/hot-glue/issues/2).
+     */
+    startTrafficCollection: (name?: string) => Promise<void>;
+    /** Stop collecting the device's traffic. This will stop the proxy on the host computer. */
+    stopTrafficCollection: () => Promise<void>;
+
+    /**
+     * Stop the app analysis and return the collected data.
+     *
+     * @param options An object with the following optional options:
+     *
+     *   - `uninstallApp`: Whether to uninstall the app after stopping the analysis (default: `false`).
+     *
+     * @returns The collected data.
+     */
+    stop: (options?: { uninstallApp?: boolean }) => Promise<AppAnalysisResult>;
+};
+
+/** The result of an app analysis. */
+export type AppAnalysisResult = {
+    /** The app's metadata. */
+    app: App;
+    /** The collected traffic, accessible by the specified name. */
+    traffic: Record<string, string>;
+};
+
+/** The options for a specific platform/run target combination. */
+// Use `unknown` here to mean "no options", and `never` to mean "not supported".
+export type RunTargetOptions<
+    Capabilities extends SupportedCapability<'android' | 'ios'>[],
+    Capability = Capabilities[number]
+> = {
+    /** The options for the Android platform. */
+    android: {
+        /** The options for the Android emulator run target. */
+        emulator: {
+            /** Options for the emulator if you want it to be automatically started and stopped by this library. */
+            startEmulatorOptions?: {
+                /** The name of the emulator to start. */
+                emulatorName: string;
+                /** Whether to start the emulator in headless mode (default: `false`). */
+                headless?: boolean;
+                /** Whether to start the emulator with audio (default: `false`). */
+                audio?: boolean;
+                /** Whether to discard all changes when exiting the emulator (default: `true`). */
+                ephemeral?: boolean;
+                /**
+                 * The proxy to use for the emulator, in the format `host:port` or `user:password@host:port`. Currently
+                 * defaults to `127.0.0.1:8080` (use an empty string to set no proxy), though the default will likely
+                 * change to "no proxy" in the future.
+                 */
+                proxy?: string;
```

Comment:
Shouldn't we get rid of this now? It's not needed anymore with WireGuard and our mitmproxy instance won't be listening on that, anyway.